app/Http/Controllers/ThirdParty/CallPowerController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Chompy\Http\Controllers\ThirdParty;
+
+use Illuminate\Http\Request;
+use Chompy\Services\Rogue;
+use Chompy\Http\Controllers\Controller;
+
+class CallPowerController extends Controller
+{
+
+    /**
+     * Create a controller instance.
+     *
+     * @return void
+     */
+    public function __construct(Rogue $rogue)
+    {
+      // @TODO: add role middleware?
+    	$this->rogue = $rogue;
+
+    }
+
+    /**
+     *
+     * @param Request $request
+     */
+    public function store(Request $request)
+    {
+    	$request->validate([
+    		// QUESTION - We shouldn't have to validate phone number because this will come from Twilio and phone numbers are validated in NS?
+            'mobile' => 'required',
+            'callpower_campaign_id' => 'required|integer',
+            // QUESTION: do we want to make it this rigid? What if CallPower statuses change?
+            'status' => 'required|string|in:completed,busy,failed,no answer,cancelled, unknown',
```

Comment:
Do we want to make sure the `status` is within `completed`, `busy`, `failed`, `no answer`, cancelled`, or `unknown` (possible CallPower statuses)? I'm leaning towards no because what if CallPower statuses change? Also, if it is anything other than `completed`, we'll just mark the post as `incomplete`, with status details in the `details` of a post. If it is `completed`, the post's `status` will be `accepted`.

Reply:
Yeah, agreed that using less validation here is probably better (since we can then always re-try the particular record in Chompy if it fails from downstream validation in Northstar or Rogue).